### PR TITLE
FEC-10370 In case No video tracks and no audio tracks  and no text tracks for player then firing Unexpected & Fatal error

### PR DESCRIPTION
**Pull request by GouravSna**

### CODE REVIEW

File: playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java [hunk 1]
```diff
@@ -124,7 +124,7 @@
 
         void onTracksOverrideABRError(PKError pkError);
         void onUnsupportedVideoTracksError(PKError pkError);
-
+        void onNoVideoAudioTrackAvailableError(PKError pkError);
```
Comment (giladna) [inline playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java]: lets change to noTracksAvaialbleError and add the text tracks to the validation if no text tracks

File: playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java [hunk 2]
```diff
@@ -158,6 +158,14 @@ boolean prepareTracks(TrackSelectionArray trackSelections) {
...
             return false;
         }
+
+        if (!checkVideoAudioTracksAvailability(mappedTrackInfo)) {
+            String errorMessage = "No Video and Audio track found";
+            PKError currentError = new PKError(PKPlayerErrorType.UNEXPECTED, PKError.Severity.Fatal, errorMessage, new IllegalStateException(errorMessage));
+            tracksErrorListener.onNoVideoAudioTrackAvailableError(currentError);
+            return false;
+        }
+
```
Comment (giladna) [inline playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java:174]: lets send an error also if the folloeing warning code

private void warnAboutUnsupportedRenderTypes() {
        if (mappedTrackInfo.getTypeSupport(TRACK_TYPE_VIDEO)
                == MappingTrackSelector.MappedTrackInfo.RENDERER_SUPPORT_UNSUPPORTED_TRACKS) {
            log.w("Warning! All the video tracks are unsupported by this device.");
        }
        if (mappedTrackInfo.getTypeSupport(TRACK_TYPE_AUDIO)
                == MappingTrackSelector.MappedTrackInfo.RENDERER_SUPPORT_UNSUPPORTED_TRACKS) {
            log.w("Warning! All the audio tracks are unsupported by this device.");
        }
    }

File: playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java (hunk 1 above)
Comment (GouravSna) [reply playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java]: Done

File: playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java [hunk 3]
```diff
@@ -1220,6 +1220,14 @@ public void onUnsupportedVideoTracksError(PKError pkError) {
                     eventListener.onEvent(PlayerEvent.Type.ERROR);
                 }
             }
+
+            @Override
+            public void noTracksAvailableError(PKError pkError) {
```
Comment (giladna) [inline playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java]: @GouravSna 
lets recheck it solves our issue

File: playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java [hunk 4]
```diff
@@ -1227,14 +1243,36 @@ private boolean isRendererTypeValid(int rendererIndex) {
...
-    private void warnAboutUnsupportedRenderTypes() {
+    private void warnAboutUnsupportedRendererTypes() {
+        int whichTrackUnsupported = -1;
+        String errorMessage;
+        PKError currentError;
+
         if (mappedTrackInfo.getTypeSupport(TRACK_TYPE_VIDEO)
                 == MappingTrackSelector.MappedTrackInfo.RENDERER_SUPPORT_UNSUPPORTED_TRACKS) {
-            log.w("Warning! All the video tracks are unsupported by this device.");
+            whichTrackUnsupported = TRACK_TYPE_VIDEO;
```
Comment (giladna) [inline playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java]: change to unsupportedTrackType?

File: playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java [hunk 5]
```diff
@@ -1227,14 +1243,36 @@ private boolean isRendererTypeValid(int rendererIndex) {
...
         }
         if (mappedTrackInfo.getTypeSupport(TRACK_TYPE_AUDIO)
                 == MappingTrackSelector.MappedTrackInfo.RENDERER_SUPPORT_UNSUPPORTED_TRACKS) {
-            log.w("Warning! All the audio tracks are unsupported by this device.");
+            if (whichTrackUnsupported == TRACK_TYPE_VIDEO) {
+                whichTrackUnsupported = 2;
```
Comment (giladna) [inline playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java]: ??2

File: playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java [hunk 6]
```diff
@@ -126,10 +126,11 @@
     }
 
     interface TracksErrorListener {
-
         void onTracksOverrideABRError(PKError pkError);
         void onUnsupportedVideoTracksError(PKError pkError);
-
+        void onUnsupportedAudioTracksError(PKError pkError);
+        void noSupportedAudioVideoTracksError(PKError pkError);
```
Comment (giladna) [inline playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java]: lets make it same as all other errors on*...

File: playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java (hunk 4 above)
Comment (giladna) [reply playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java]: rename

File: playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java (hunk 5 above)
Comment (GouravSna) [reply playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java]: Removed it.

File: playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java (hunk 4 above)
Comment (GouravSna) [reply playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java]: Done

File: playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java (hunk 2 above)
Comment (GouravSna) [reply playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java:174]: Done

File: playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java (hunk 6 above)
Comment (GouravSna) [reply playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java]: Done

File: playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java (hunk 3 above)
Comment (GouravSna) [reply playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java]: Unit tested it. It is sending the the proper error events.

**Review by giladna — APPROVED** [review]

**Pull request merged**